scripts/1-fetch/arxiv_fetch.py

```diff
@@ -0,0 +1,643 @@
+
+# Third-party
+import feedparser
+import requests
+import yaml
+from pygments import highlight
+from pygments.formatters import TerminalFormatter
+from pygments.lexers import PythonTracebackLexer
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+# Add parent directory so shared can be imported
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+# Add dev directory for category converter
+sys.path.append(os.path.join(os.path.dirname(__file__), "..", "..", "dev"))
+
+# Third-party
+import arxiv_category_converter  # noqa: E402
+
+# First-party/Local
+import shared  # noqa: E402
+
+# Setup
+LOGGER, PATHS = shared.setup(__file__)
+
+# Constants
+BASE_URL = "http://export.arxiv.org/api/query?"
+FILE_ARXIV_COUNT = shared.path_join(PATHS["data_1-fetch"], "arxiv_1_count.csv")
+FILE_ARXIV_CATEGORY = shared.path_join(
+    PATHS["data_1-fetch"], "arxiv_2_count_by_category.csv"
+)
+FILE_ARXIV_CATEGORY_REPORT = shared.path_join(
+    PATHS["data_1-fetch"], "arxiv_2_count_by_category_report.csv"
+)
+FILE_ARXIV_CATEGORY_REPORT_AGGREGATE = shared.path_join(
+    PATHS["data_1-fetch"], "arxiv_2_count_by_category_report_agg.csv"
+)
+FILE_ARXIV_YEAR = shared.path_join(
+    PATHS["data_1-fetch"], "arxiv_3_count_by_year.csv"
+)
+FILE_ARXIV_AUTHOR = shared.path_join(
+    PATHS["data_1-fetch"], "arxiv_4_count_by_author_count.csv"
+)
+FILE_ARXIV_AUTHOR_BUCKET = shared.path_join(
+    PATHS["data_1-fetch"], "arxiv_4_count_by_author_bucket.csv"
+)
+# records metadata for each run for audit, reproducibility, and provenance
+FILE_PROVENANCE = shared.path_join(PATHS["data"], "arxiv_provenance.yaml")
+
+HEADER_COUNT = ["TOOL_IDENTIFIER", "COUNT"]
+HEADER_CATEGORY = ["TOOL_IDENTIFIER", "CATEGORY", "COUNT"]
+HEADER_CATEGORY_REPORT = [
+    "TOOL_IDENTIFIER",
+    "CATEGORY_CODE",
+    "CATEGORY_LABEL",
+    "COUNT",
+    "PERCENT",
+]
+HEADER_YEAR = ["TOOL_IDENTIFIER", "YEAR", "COUNT"]
+HEADER_AUTHOR = ["TOOL_IDENTIFIER", "AUTHOR_COUNT", "COUNT"]
+HEADER_AUTHOR_BUCKET = ["TOOL_IDENTIFIER", "AUTHOR_BUCKET", "COUNT"]
+
+QUARTER = os.path.basename(PATHS["data_quarter"])
+
+CATEGORY_LABELS = {}
+
+# Compiled regex patterns for CC license detection
+CC_PATTERNS = [
+    (re.compile(r"\bCC[-\s]?0\b", re.IGNORECASE), "CC0"),
+    (
+        re.compile(r"\bCC[-\s]?BY[-\s]?NC[-\s]?ND\b", re.IGNORECASE),
+        "CC BY-NC-ND",
+    ),
+    (
+        re.compile(r"\bCC[-\s]?BY[-\s]?NC[-\s]?SA\b", re.IGNORECASE),
+        "CC BY-NC-SA",
+    ),
+    (re.compile(r"\bCC[-\s]?BY[-\s]?ND\b", re.IGNORECASE), "CC BY-ND"),
+    (re.compile(r"\bCC[-\s]?BY[-\s]?SA\b", re.IGNORECASE), "CC BY-SA"),
+    (re.compile(r"\bCC[-\s]?BY[-\s]?NC\b", re.IGNORECASE), "CC BY-NC"),
+    (re.compile(r"\bCC[-\s]?BY\b", re.IGNORECASE), "CC BY"),
+    (
+        re.compile(r"\bCREATIVE\s+COMMONS\b", re.IGNORECASE),
+        "UNKNOWN CC legal tool",
+    ),
+]
+
+# Log the start of the script execution
+LOGGER.info("Script execution started.")
```

Comment:
You should put this in the main function so it triggers when the script actually runs